valhalla/sif/dynamiccost.h

```diff
@@ -279,6 +281,7 @@ class DynamicCost {
                   const baldr::GraphTile*& tile,
                   const baldr::GraphId& edgeid,
                   const bool forward,
+                  thor::EdgeStatus* edgestatus = nullptr,
```

Comment:
I'm not overly happy with this solution as this ties the implementation of the a-star algorithms to the costing function.

The reason I did this though was that I absolutely wanted to avoid duplicating the complex logic for walking the graph and comparing to restrictions. Perhaps a better solution would have been to send a callback function as parameter to tweak the `edgestatus` variables in the a-stars...